test/unit/ngsiv2/HTTP_reveice_measures-test.js

```diff
@@ -0,0 +1,329 @@
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ *
+ * Modified by: Daniel Calvo - ATOS Research & Innovation
+ */
+
+/* jshint camelcase: false */
+
+'use strict';
+
+var iotagentMqtt = require('../../../'),
+    config = require('./config-test.js'),
+    nock = require('nock'),
+    iotAgentLib = require('iotagent-node-lib'),
+    should = require('should'),
+    async = require('async'),
+    request = require('request'),
+    utils = require('../../utils'),
+    groupCreation = {
+        url: 'http://localhost:' + config.iota.server.port + '/iot/services',
+        method: 'POST',
+        json: {
+            services: [
+                {
+                    resource: '/iot/json',
+                    apikey: 'KL223HHV8732SFL1',
+                    entity_type: 'TheLightType',
+                    trust: '8970A9078A803H3BL98PINEQRW8342HBAMS',
+                    cbHost: 'http://unexistentHost:1026',
+                    commands: [],
+                    lazy: [],
+                    attributes: [
+                        {
+                            name: 'status',
+                            type: 'Boolean'
+                        }
+                    ],
+                    static_attributes: []
+                }
+            ]
+        },
+        headers: {
+            'fiware-service': 'smartGondor',
+            'fiware-servicepath': '/gardens'
+        }
+    },
+    contextBrokerMock;
+
+describe('HTTP: Measure reception ', function() {
+    beforeEach(function(done) {
+        var provisionOptions = {
+            url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/deviceProvisioning/provisionDeviceHTTP.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        contextBrokerMock = nock('http://192.168.1.1:1026')
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v1/updateContext')
+            .reply(200, utils.readExampleFile('./test/contextResponses/multipleMeasuresSuccess.json'));
+
+        iotagentMqtt.start(config, function() {
+            request(provisionOptions, function(error, response, body) {
+                done();
+            });
+        });
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+
+        async.series([
+            iotAgentLib.clearAll,
+            iotagentMqtt.stop
+        ], done);
+    });
+
+    describe('When a POST measure arrives for the HTTP binding', function() {
+        var optionsMeasure = {
+            url: 'http://localhost:' + config.http.port + '/iot/json',
+            method: 'POST',
+            json: {
```

Comment:
It would be great to include some tests for JSON native types different from string, eg:

```
            json: {
                humidity: 32,
                active: true
            },
```

in order to check if the IOTAgent is able to push them with the correct native type to CB. Note this is a new possiblity with NGSIv2 that the agent hasn't needed to consider before (as with NGSIv1 no matter if the agent was able to push native types, they get "stringfied" by Orion upon reception time).

Reply:
(Maybe the PR include tests like that but I haven't reach them yet in my review... however, I'm writting the comment now so I don't forget :)

Reply:
Fixed with ccbe539170af3427946f75901b6f8e5540f158ad